userspace/libscap/scap_udig.c

```diff
@@ -147,43 +492,46 @@ int32_t udig_alloc_ring_descriptors(int* ring_descs_fd,
 {
 	uint32_t mem_size = sizeof(struct ppm_ring_buffer_info) + sizeof(struct udig_ring_buffer_status);
 
-	//
-	// First, try to open an existing ring
-	//
+#ifdef UDIG
+	// This is a producer, let's tell it to open an existing
+	// ring using ud_shm_open
```

Comment:
that's what the last parameter equal to zero means? at this point I believe producer_open and consumer_open should be two separate functions

Reply:
I reverted all of this now while you were commenting,
everything is in `udig_memfd_shm_open` and all the changes to `udig_alloc_ring*` functions are not needed anymore.